install-pip-dependencies.sh

```diff
@@ -11,6 +11,9 @@
 
 set -o nounset -o errexit
 
+# Remove the pip cache so we are not installing stale packages:
+pip cache purge || true
```

Comment:
Also, it runs afoul of the rule that (IMO) an install script should only make changes in its own subdirectory (and not `~/.cache/pip`) unless explicitly requested. Maybe there's an option to tell pip to ignore its cache?